lib/LaTeXML/resources/XSLT/LaTeXML-inline-xhtml.xsl

```diff
@@ -33,22 +33,41 @@
   <xsl:preserve-space elements="ltx:text"/>
   <xsl:template match="ltx:text">
     <xsl:param name="context"/>
-    <xsl:element name="span" namespace="{$html_ns}">
-      <xsl:variable name="innercontext" select="'inline'"/><!-- override -->
-      <xsl:call-template name="add_id"/>
-      <xsl:call-template name="add_attributes">
-        <xsl:with-param name="extra_classes" select="f:if(@width,'ltx_inline-block','')"/>
-      </xsl:call-template>
-      <xsl:apply-templates select="." mode="begin">
-        <xsl:with-param name="context" select="$innercontext"/>
-      </xsl:apply-templates>
-      <xsl:apply-templates>
-        <xsl:with-param name="context" select="$innercontext"/>
-      </xsl:apply-templates>
-      <xsl:apply-templates select="." mode="end">
-        <xsl:with-param name="context" select="$innercontext"/>
-      </xsl:apply-templates>
-    </xsl:element>
+    <xsl:variable name="content">
```

Comment:
Just proof of concept, this would have to be implemented for ltx:inline-block and ltx:p too (the other possible constructors from `\hbox`).